tools/scripts/upload-ab-to-s3.js

```diff
@@ -0,0 +1,133 @@
+
+var argv = require('yargs')
+  .usage('$0 [options]')
+
+  .string('env')
+  .alias('env', 'environment')
+  .describe('env', 'NR Internal Environment')
+  .default('env', 'dev')
+
+  .string('current')
+  .describe('current', 'current/stable build (defaults to -current)')
+  .default('current', 'https://js-agent.newrelic.com/nr-loader-spa-current.min.js')
+
+  .string('next')
+  .describe('next', 'next version to compare to stable version (defaults to /dev)')
+  .default('next', 'https://js-agent.newrelic.com/dev/nr-loader-spa.min.js')
+
+  .string('config')
+  .describe('config', 'NR internal config block')
+
+  .string('bucket')
+  .describe('bucket', 'S3 bucket name')
+
+  .string('role')
+  .describe('role', 'S3 role ARN')
+
+  .boolean('dry')
+  .default('dry', false)
+  .argv
+
+
+const { env, config, bucket, role, current, next, dry } = argv
+
+let counter = 0
+
+if (!env || !config || !bucket || !role) {
+  console.log("missing required param")
+  process.exit(1)
+}
+
+(async function () {
+  const filePaths = [
+    current,
+    next,
+    ...(env === 'dev' && await getOpenPrNums()).map(num => `https://js-agent.newrelic.com/pr/PR${num}`),
+  ]
+
+  Promise.all(filePaths.map(fp => getFile(fp))).then((contents) => {
+    contents = contents.filter(([url, res])=> res.statusCode === 200)
+    console.log(`found ${contents.length} valid PR builds in CDN`)
+    let output = `window.NREUM=${config};`
+    contents.forEach(([url, res, body]) => { output += wrapAgent(body) })
+    output += randomExecutor(contents.length)
+
+    const filename = `internal/${env}.js`
+
+    connectToS3(role, dry).then(async () => {
+      const uploads = await uploadToS3(filename, output, bucket, dry)
+      console.log(`Successfully uploaded ${filename} to S3`)
+      process.exit(0)
+    }).catch(err => {
+      console.log(err)
+      process.exit(1)
+    })
+  }).catch(err => {
+    console.log("error getting all files... ", err)
+  })
+
+})()
+
+function wrapAgent(agent) {
+  const agentNo = counter++
+  return `
+        function agent${agentNo}(){
+            ${agent}
+        }
+    `
+}
+
+function randomExecutor(fnCount) {
+  let output = `
+        (function (){
+            var r = Math.random();
+    `
+  for (var i = 0; i < fnCount; i++) {
+    output += `if (r < ${((i + 1) / fnCount) + 0.00001}) return agent${i}();\n`
+  }
+  output += '})()'
+  return output
+}
+
+function getFile(path) {
+  var opts = {
+    uri: path,
+    method: 'GET',
+    gzip: true
+  }
+
+  console.log('downloading ', path)
+
+  return new Promise((resolve, reject) => {
+    request(opts, (err, res, body) => {
+      if (err) {
+        reject(err)
+        return
+      }
+      resolve([path, res, body])
+    })
+  })
+}
+
+function getOpenPrNums() {
+  var opts = {
+    uri: 'https://api.github.com/repos/newrelic/newrelic-browser-agent/pulls?state=open',
+    method: 'GET',
+    headers: {
+      'User-Agent': 'metal-messiah'
```

Comment:
Is this the UA string? Or is it supposed to be a gh user? If latter, there's a browser agent collective user I think

Reply:
Ah this is a remnant from local testing, TY